TestAntctlAgentLocalAccess failed occasionally

**Describe the bug**
`TestAntctlAgentLocalAccess` failed occasionally.
```
--- FAIL: TestAntctlAgentLocalAccess (18.54s)
    fixtures.go:55: Creating K8s clientset
    fixtures.go:61: Creating 'antrea-test' K8s Namespace
    fixtures.go:38: Applying Antrea YAML
    fixtures.go:42: Waiting for all Antrea DaemonSet Pods
    fixtures.go:46: Checking CoreDNS deployment
    antctl_test.go:14: antctl stdout:
    antctl_test.go:15: antctl stderr:
        I0224 02:08:27.755039      14 command_definition.go:440] Requesting URI /apis/system.antrea.tanzu.vmware.com/v1beta1/version
        I0224 02:08:27.755053      14 client.go:112] antctl runs as local mode
        Error: Get http://antctl/apis/system.antrea.tanzu.vmware.com/v1beta1/version: dial unix /var/run/antctl.sock: connect: no such file or directory
        Usage:
          antctl version [flags]
        
        Examples:
          Get the transformedversion
          $ antctl version
        
        
        Flags:
          -h, --help            help for version
          -o, --output string   output format: json|yaml|table (default "json")
        
        Global Flags:
          -k, --kubeconfig string   absolute path to the kubeconfig file (default "/root/.kube/config")
          -t, --timeout duration    time limit of the execution of the command
          -v, --verbose             enable verbose output
        
    antctl_test.go:45: Error when running `antctl version` from antrea-agent-t29zf: command terminated with exit code 1
    fixtures.go:89: Exporting test logs to '/var/lib/jenkins/workspace/antrea-automation-e2e-for-pull-request/antrea-test-logs/TestAntctlAgentLocalAccess.Feb24-02-08-27'
    fixtures.go:190: Error when exporting kubelet logs: error when running journalctl on Node 'antrea-networkpolicy2-0', is it available?
    fixtures.go:196: Deleting 'antrea-test' K8s Namespace
```
agent log shows the antctl server was started at 02:08:27.823677 after the test sent CLI request (02:08:27.755053):
```
I0224 02:08:26.626234       1 agent.go:51] Starting Antrea agent (version v0.5.0-dev-81be4c6)
I0224 02:08:26.626276       1 client.go:33] No kubeconfig file was specified. Falling back to in-cluster config
I0224 02:08:26.632836       1 client.go:23] No antrea kubeconfig file was specified. Falling back to in-cluster config
I0224 02:08:26.634113       1 ovs_client.go:65] Connecting to OVSDB at address /run/openvswitch/db.sock
I0224 02:08:27.634309       1 ovs_client.go:84] Not connected yet, will try again in 2s
I0224 02:08:27.636244       1 agent.go:190] Setting up node network
I0224 02:08:27.791505       1 ovs_client.go:106] Bridge exists: bb499a48-b5a8-4c70-89c5-c7b51635fdcb
I0224 02:08:27.793593       1 agent.go:563] Using round number 44793
I0224 02:08:27.797984       1 ofctrl_bridge.go:178] OFSwitch is connected: 00:00:4a:9a:49:bb:70:4c
I0224 02:08:27.809518       1 agent.go:218] Agent initialized NodeConfig=NodeName: antrea-networkpolicy2-0, PodCIDR: 192.168.0.0/24, NodeIP: 10.176.25.24/22, Gateway: Name gw0: IP 192.168.0.1, MAC 9e:d4:64:8f:74:3d, NetworkConfig=&{Encap vxlan false }
I0224 02:08:27.812972       1 server.go:528] Reconciliation for CNI server
I0224 02:08:27.822896       1 server.go:499] Starting CNI server
I0224 02:08:27.823465       1 command_list.go:61] Adding cli handler /apis/system.antrea.tanzu.vmware.com/v1beta1/version
I0224 02:08:27.823560       1 command_list.go:61] Adding cli handler /apis/system.antrea.tanzu.vmware.com/v1beta1/agent-info
I0224 02:08:27.823677       1 server.go:68] Starting antctl server
```
The test should ensure antrea-agent is ready before testing its API.
We can leverage readiness probe, which is currently a dummy script:
https://github.com/vmware-tanzu/antrea/blob/269299f678971e421a4493b703469ecc5203c887/build/images/scripts/container_liveness_probe#L5-L6

**Expected**
The test should succeed consistently,

**Actual behavior**
It failed occasionally.

**Versions:**
Please provide the following information:
 - Antrea version (Docker image tag).
 - Kubernetes version (use `kubectl version`). If your Kubernetes components have [different versions](https://kubernetes.io/docs/setup/release/version-skew-policy/), please provide the version for all of them.
 - Container runtime: which runtime are you using (e.g. containerd, cri-o, docker) and which version are you using?
 - Linux kernel version on the Kubernetes Nodes (`uname -r`).
 - If you chose to compile the Open vSwitch kernel module manually instead of using the kernel module built into the Linux kernel, which version of the OVS kernel module are you using? Include the output of `modinfo openvswitch` for the Kubernetes Nodes.

**Additional context**
Add any other context about the problem here, such as Antrea logs, kubelet logs, etc.

(Please consider pasting long output into a [GitHub gist](https://gist.github.com) or any other pastebin.)

--- a/build/yamls/antrea-ipsec.yml
+++ b/build/yamls/antrea-ipsec.yml
@@ -371,6 +371,16 @@ spec:
         ports:
         - containerPort: 443
           protocol: TCP
+        readinessProbe:
+          failureThreshold: 5
+          httpGet:
+            host: 127.0.0.1
+            path: /healthz
+            port: 443
+            scheme: HTTPS
+          initialDelaySeconds: 5
+          periodSeconds: 10
+          timeoutSeconds: 5
         volumeMounts:
         - mountPath: /etc/antrea/antrea-controller.conf
           name: antrea-config
@@ -490,6 +500,16 @@ spec:
           periodSeconds: 10
           timeoutSeconds: 5
         name: antrea-agent
+        readinessProbe:
+          failureThreshold: 5
+          httpGet:
+            host: 127.0.0.1
+            path: /healthz
+            port: 10443
+            scheme: HTTPS
+          initialDelaySeconds: 5
+          periodSeconds: 10
+          timeoutSeconds: 5
         securityContext:
           privileged: true
         volumeMounts:

--- a/build/yamls/antrea.yml
+++ b/build/yamls/antrea.yml
@@ -362,6 +362,16 @@ spec:
         ports:
         - containerPort: 443
           protocol: TCP
+        readinessProbe:
+          failureThreshold: 5
+          httpGet:
+            host: 127.0.0.1
+            path: /healthz
+            port: 443
+            scheme: HTTPS
+          initialDelaySeconds: 5
+          periodSeconds: 10
+          timeoutSeconds: 5
         volumeMounts:
         - mountPath: /etc/antrea/antrea-controller.conf
           name: antrea-config
@@ -449,6 +459,16 @@ spec:
           periodSeconds: 10
           timeoutSeconds: 5
         name: antrea-agent
+        readinessProbe:
+          failureThreshold: 5
+          httpGet:
+            host: 127.0.0.1
+            path: /healthz
+            port: 10443
+            scheme: HTTPS
+          initialDelaySeconds: 5
+          periodSeconds: 10
+          timeoutSeconds: 5
         securityContext:
           privileged: true
         volumeMounts:

--- a/build/yamls/base/agent.yml
+++ b/build/yamls/base/agent.yml
@@ -87,6 +87,16 @@ spec:
             timeoutSeconds: 5
             periodSeconds: 10
             failureThreshold: 5
+          readinessProbe:
+            httpGet:
+              host: 127.0.0.1
+              path: /healthz
+              port: 10443
+              scheme: HTTPS
+            initialDelaySeconds: 5
+            timeoutSeconds: 5
+            periodSeconds: 10
+            failureThreshold: 5
           securityContext:
             # antrea-agent needs to manipulate /proc/sys/net/ipv4/conf/XXX/send_redirects"
             privileged: true

--- a/build/yamls/base/controller.yml
+++ b/build/yamls/base/controller.yml
@@ -80,6 +80,16 @@ spec:
           ports:
             - containerPort: 443
               protocol: TCP
+          readinessProbe:
+            httpGet:
+              host: 127.0.0.1
+              path: /healthz
+              port: 443
+              scheme: HTTPS
+            initialDelaySeconds: 5
+            timeoutSeconds: 5
+            periodSeconds: 10
+            failureThreshold: 5
           volumeMounts:
             - name: antrea-config
               mountPath: /etc/antrea/antrea-controller.conf

